core/templates/domain/exploration/editable-exploration-backend-api.service.ts

```diff
@@ -15,152 +15,170 @@
-          var exploration = angular.copy(response.data);
-
-          // Delete from the ReadOnlyExplorationBackendApiService's cache
-          // As the two versions of the data (learner and editor) now differ
-          ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
-            explorationId, exploration);
-
-          if (successCallback) {
-            successCallback(exploration);
-          }
-        }, function(errorResponse) {
-          if (errorCallback) {
-            errorCallback(errorResponse.data);
-          }
-        }
-      );
-    };
+  private _updateExploration(
+      explorationId: string, explorationVersion: string,
+      commitMessage: string, changeList: Array<string>,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var editableExplorationDataUrl = this._getExplorationUrl(
+      explorationId, null);
 
-    var _deleteExploration = function(
-        explorationId, successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(explorationId, null);
+    var putData = {
+      version: explorationVersion,
+      commit_message: commitMessage,
+      change_list: changeList
+    };
+    this.http.put(editableExplorationDataUrl, putData).toPromise().then(
+      (response) => {
+        // The returned data is an updated exploration dict.
+        var exploration = cloneDeep(response);
 
-      $http['delete'](editableExplorationDataUrl).then(function() {
-        // Delete item from the ReadOnlyExplorationBackendApiService's cache
-        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+        // Delete from the ReadOnlyExplorationBackendApiService's cache
+        // As the two versions of the data (learner and editor) now differ
+        this.readOnlyExplorationBackendApiService.deleteExplorationFromCache(
           explorationId);
+
         if (successCallback) {
-          successCallback({});
+          successCallback(exploration);
         }
-      }, function(errorResponse) {
+      }, (errorResponse) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse.error);
         }
-      });
-    };
+      }
+    );
+  }
 
-    var _getExplorationUrl = function(explorationId, applyDraft) {
-      if (applyDraft) {
-        return UrlInterpolationService.interpolateUrl(
-          EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE, {
-            exploration_id: explorationId,
-            apply_draft: JSON.stringify(applyDraft)
-          }
-        );
+  private _deleteExploration(
+      explorationId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var editableExplorationDataUrl = this._getExplorationUrl(
+      explorationId, null);
+    // eslint-disable-next-line
```

Comment:
Why do we need this?

Reply:
I belive because delete is a javascript keyword and eslint will keep warning since it's a delete request. @kevintab95

Reply:
Ah I see. Thanks for explaining!